src/Gui/Placement.cpp

```diff
@@ -178,33 +178,33 @@ void Placement::openTransaction()
 QWidget* Placement::getInvalidInput() const
 {
     QList<Gui::QuantitySpinBox*> sb = this->findChildren<Gui::QuantitySpinBox*>();
-    for (QList<Gui::QuantitySpinBox*>::iterator it = sb.begin(); it != sb.end(); ++it) {
-        if (!(*it)->hasValidInput())
-            return (*it);
+    for (const auto & it : sb) {
+        if (!it->hasValidInput())
+            return it;
     }
     return nullptr;
 }
 
 void Placement::revertTransformation()
 {
-    for (std::set<std::string>::iterator it = documents.begin(); it != documents.end(); ++it) {
-        Gui::Document* document = Application::Instance->getDocument(it->c_str());
+    for (const auto & it : documents) {
+        Gui::Document* document = Application::Instance->getDocument(it.c_str());
         if (!document) continue;
 
         if (!changeProperty) {
             std::vector<App::DocumentObject*> obj = document->getDocument()->
                 getObjectsOfType(App::DocumentObject::getClassTypeId());
             if (!obj.empty()) {
-                for (std::vector<App::DocumentObject*>::iterator it=obj.begin();it!=obj.end();++it) {
+                for (const auto & it : obj) {

```

Comment:
Let's rename this one to avoid any confusion.

Reply:
I would be cheeky and say that's being selectively pedantic and doesn't relate directly to auto. If `obj` more correctly had a plural name e.g. `objs`, then `obj` would have been used by IDE. There are many thousands of `it` and `jt`, and singular names for plural objects  in the code base. A number of times during this PR I found myself undoing the IDE and resetting to `it` to keep (you) reviewer Gods happy!

Reply:
For reference, my particular objection to this one is that it redefines `it` from the outer scope. Overall I'd prefer to use better loop variable names, and in lots of cases you did that, when already changing the loop anyway (which I thank you for).

Reply:
Yes, the variable shadows another, like thousands of other times in this code base. Most of the ones I did redefine were done by the IDE. My point is that I'm now too scared to deliberately do any drive-by fixes lest the PR be knocked back again. I could do a separate PR chasing these in future maybe.